include/mls/messages.h

```diff
@@ -455,12 +480,52 @@ struct Sender
   TLS_SERIALIZABLE(sender_type, sender)
 };
 
+struct MLSPlaintextTBS
+{
+  const var::variant<GroupContext>& context;
+
+  const bytes& group_id;
+  const epoch_t& epoch;
+  const Sender& sender;
+  const bytes& authenticated_data;
+
+  const ContentType& content_type;
+  const var::variant<ApplicationData, Proposal, Commit>& content;
+
+  TLS_SERIALIZABLE(context,
+                   group_id,
+                   epoch,
+                   sender,
+                   authenticated_data,
+                   content_type,
+                   content)
+  TLS_TRAITS(tls::variant<SenderType>,
+             tls::vector<1>,
+             tls::pass,
+             tls::pass,
+             tls::vector<4>,
+             tls::pass,
+             tls::variant<ContentType>)
+};
+
+struct MLSPlaintextTBM
+{
+  const MLSPlaintextTBS& tbs;
+  const bytes& signature;
+  const std::optional<MAC>& confirmation_tag;
+
+  TLS_SERIALIZABLE(tbs, signature, confirmation_tag)
+  TLS_TRAITS(tls::pass, tls::vector<2>, tls::pass)
+};
+
 struct MLSPlaintext
 {
   bytes group_id;
   epoch_t epoch;
   Sender sender;
   bytes authenticated_data;
+
+  ContentType content_type;
```

Comment:
Is there a reason to make this explicit, as opposed to having it as part of the `variant`?

Reply:
At the moment it is explicit in the [draft](https://messaginglayersecurity.rocks/mls-protocol/draft-ietf-mls-protocol.html#section-9), so I thought it might lead to interop issues.

Reply:
Correction for posterity: in mlspp, the pattern `T type; select (Obj.type)` is directly implemented in the serialisation of `var::variant`, so it does not have to be a part of the struct. What is currently not spec-compliant is any pattern `select (Obj.type)` without the preceding tag.